server/routes/utils.js

```diff
@@ -58,6 +59,30 @@ async function getTeamDocument(req, res, next) {
     return next()
 }
 
+// this middleware checks if the given teamIds are actual documents, and if so, populate them
+// it will also check if the given number of rounds fit can fit within a season
+async function _validateFixture(req, res, next) {
+    const { dateStart, dateFinish } = req.season
+    // Check valid teams
+    if (!(await allValidDocumentIds(req.body.teamIds, Team))) {
+        return res.status(404).json({ success: false, error: 'Some team does not exist' })
+    }
+    // Check team is added to grade
+    req.body.teamIds.forEach(team => {
+        if (!req.grade.teams.includes(team)) {
+            console.log(team)
+            return res.status(400).json({ success: false, error: 'Team is not added to grade' })
+        }
+    })
+    // Check number of rounds can fit within season
+    if (dateStart.setDate(dateStart.getDate() + req.body.numRounds * 7) > dateFinish) {
```

Comment:
might be worth to have some basic error handling for when `numRounds` and `datesAndLocations` aren't given. atm it just gives a message like "cannot read property length of undefined" once it reaches the controller.

maybe also requiring `req.body.numRounds > 0` would be good, otherwise the controller returns successfully but just doesn't do anything

Reply:
also good point, forgot to check the body params being passed in by the user so will have a check for these asw